include/zephyr/drivers/video.h

```diff
@@ -818,8 +818,9 @@ static inline uint64_t video_frmival_nsec(const struct video_frmival *frmival)
  * @param stepwise The stepwise frame interval range to search
  * @param desired The frame interval for which find the closest match
  * @param match The resulting frame interval closest to @p desired
+ * @return 0 on success, -EINVAL if step is 0 or input invalid
  */
-void video_closest_frmival_stepwise(const struct video_frmival_stepwise *stepwise,
+int  video_closest_frmival_stepwise(const struct video_frmival_stepwise *stepwise,
```

Comment:
Little concerned by this API change in a public header, even if it is minor. Is it possible to fix this without modifying the API for this release? Or is this something we have to get done now?

Reply:
A division by zero would only happen when the user requests infinite FPS or zero FPS.

- If the caller does parameter validation then it might be an `__ASSERT_NO_MSG()` I guess.
- If the caller relies on it for parameter validation then it might need the `return` to check.

Glad to know the most "zephyr way" to do it to keep APIs uniform across areas.

Reply:
I am ok with `__ASSERT_NO_MSG`- also ok with logging an error and returning, would either of these work? To be clear I'm totally ok with this API changing in 4.3- just want to keep things stable for this release since we are already in the RC phase

Reply:
No pressure in this case, so may not be worth the risks of rushing it in 4.2.
Thank you for the vigilance!